src/Agent.Sdk/SecretMasking/LegacyLoggedSecretMasker.cs

```diff
@@ -129,13 +136,34 @@ public ISecretMasker Clone()
         return new LegacyLoggedSecretMasker(this._secretMasker.Clone());
     }
 
+    public double elapsedMaskingTime;
+
+
     public string MaskSecrets(string input)
     {
-        return this._secretMasker.MaskSecrets(input);
+        var stopwatch = Stopwatch.StartNew();
+        string result = this._secretMasker.MaskSecrets(input);
+        Interlocked.Exchange(ref elapsedMaskingTime, stopwatch.ElapsedTicks);
+        return result;
     }
 
     Sdk.SecretMasking.ISecretMasker Sdk.SecretMasking.ISecretMasker.Clone()
     {
         return new LegacyLoggedSecretMasker(this._secretMasker.Clone());
     }
+
+    public IDictionary<string, string> GetTelemetry()
```

Comment:
>GetTelemetry [](http://example.com/codeflow?start=39&length=12)

Note that we only produce the `ElapsedMaskingTime` data point for the legacy masker, as I think we are only really concerned about relative runtime perf differences between the maskers (i.e., and other telemetry isn't interesting/won't be consumed for the legacy masker).